### PR TITLE
Ruby 3

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: config/deploy/docker/assets/entrypoint.sh [hunk 1]
```diff
@@ -63,4 +63,4 @@ cd ../..
 
 # Then exec the container's main process (what's set as CMD in the Dockerfile).
 echo "calling: $@"
-exec "$@"
+bundle exec "$@"
```
Comment (yakloinsteak) [inline config/deploy/docker/assets/entrypoint.sh:66]: Did you test this? the running process in the container will now be in a subprocess.

File: config/initializers/protected_id.rb [hunk 2]
```diff
@@ -67,6 +69,17 @@ def deobfuscate(slug)
       [id_part, day_stamp]
     end
     module_function :deobfuscate
+
+    def des_ecb_decrypt(encrypted)
```
Comment (yakloinsteak) [inline config/initializers/protected_id.rb]: Did you get this to work?

File: config/initializers/psych.rb [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
-                     |[-+]?0[0-7_,]+                         (?# base 8)
-                     |[-+]?(?:0|[1-9](?:[0-9]|,[0-9]|_[0-9])*) (?# base 10)
-                     |[-+]?0x[0-9a-fA-F_,]+                  (?# base 16))$/x
-    else
-      puts 'Check Psych monkeypatch!'
-    end
-  end
-end
\ No newline at end of file
+# module Psych
```
Comment (yakloinsteak) [inline config/initializers/psych.rb]: Why are we keeping this file?

File: docker/Dockerfile [hunk 4]
```diff
@@ -39,10 +39,9 @@ RUN apk add --no-cache \
   vim less \
   tar xz p7zip \
   python3 py3-pip \
-  && git config --global --add safe.directory /app \
-  && ln -s $(ls -1r /usr/lib/libproj.so.* | head -1) /usr/lib/libproj.so.25
+  && git config --global --add safe.directory /app
```
Comment (yakloinsteak) [inline docker/Dockerfile:42]: Did you test any postgis functionality?

File: docker/shell.entrypoint.sh [hunk 5]
```diff
@@ -1,24 +1,5 @@
 #!/bin/bash
 set -e
```
Comment (yakloinsteak) [inline docker/shell.entrypoint.sh]: If you took this entrypoint out of the docker compose and docker file, then I think we don't need it at all.

**Review by yakloinsteak — COMMENTED** [review]
Looks fine. I'd approve but I want to make sure that `exec` becoming `bundle exec` works correctly.

File: config/initializers/psych.rb (hunk 3 above)
Comment (awisema) [reply config/initializers/psych.rb]: Or, inversely, why were we getting YAML files will trailing underscores, and did we stop?

**Review by awisema — APPROVED** [review]

File: config/deploy/docker/assets/entrypoint.sh (hunk 1 above)
Comment (eanders) [reply config/deploy/docker/assets/entrypoint.sh:66]: I couldn't get it to work without that.  I think the issue is that `$@` is `rails asset:compile` or something similar, which needed the bundle.  I suppose we could change what gets passed in to include the `bundle`

File: config/initializers/protected_id.rb (hunk 2 above)
Comment (eanders) [reply config/initializers/protected_id.rb]: Nope, just forgot to remove the gem and code.  Thanks!

File: config/initializers/psych.rb (hunk 3 above)
Comment (eanders) [reply config/initializers/psych.rb]: Only because I wasn't 100% sure it was working, but it is, so file removed.

File: docker/Dockerfile (hunk 4 above)
Comment (eanders) [reply docker/Dockerfile:42]: No, but we definitely should.  Thanks for the reminder.

File: docker/shell.entrypoint.sh (hunk 5 above)
Comment (eanders) [reply docker/shell.entrypoint.sh]: correct!

File: config/deploy/docker/assets/entrypoint.sh (hunk 1 above)
Comment (yakloinsteak) [reply config/deploy/docker/assets/entrypoint.sh:66]: I'd try changing that to `./bin/rails asset:compile` instead if possible. Though, if this doesn't break anything it's fine as is.

**Pull request merged**